From 6e18026c88642af52b0f17a5368d104b7aa57b5a Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Mon, 26 Jun 2017 16:10:30 +0800
Subject: [PATCH 2006/5242] MLK-15285-2 soc: imx: add HAVE_IMX_SOC for soc
 driver

commit  06b40e6019ab250a27beb5d1fd8cd169d1dbd5ea from
https://source.codeaurora.org/external/imx/linux-imx.git

soc driver is NOT only for i.MX8QM, add HAVE_IMX_SOC
config to improve build dependency.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/Kconfig  |    6 ++++++
 drivers/soc/imx/Makefile |    3 ++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/imx/Kconfig b/drivers/soc/imx/Kconfig
index 3daacb6..c9c7022 100644
--- a/drivers/soc/imx/Kconfig
+++ b/drivers/soc/imx/Kconfig
@@ -12,16 +12,19 @@ config ARCH_FSL_IMX8QM
 	bool "i.MX8QM"
 	select HAVE_IMX_MU
 	select HAVE_IMX_RPMSG
+	select HAVE_IMX_SOC
 
 config ARCH_FSL_IMX8QXP
 	bool "i.MX8QXP"
 	select HAVE_IMX_MU
 	select HAVE_IMX_RPMSG
+	select HAVE_IMX_SOC
 
 config ARCH_FSL_IMX8MQ
 	bool "i.MX8MQ"
 	select HAVE_IMX_MU
 	select HAVE_IMX_RPMSG
+	select HAVE_IMX_SOC
 
 endif
 
@@ -35,6 +38,9 @@ config IMX7_PM_DOMAINS
 config HAVE_IMX_MU
 	bool
 
+config HAVE_IMX_SOC
+	bool
+
 config HAVE_IMX_RPMSG
 	select RPMSG_VIRTIO
 	select RPMSG
diff --git a/drivers/soc/imx/Makefile b/drivers/soc/imx/Makefile
index 6668f62..2a12048 100644
--- a/drivers/soc/imx/Makefile
+++ b/drivers/soc/imx/Makefile
@@ -2,5 +2,6 @@ obj-$(CONFIG_HAVE_IMX_GPC) += gpc.o
 obj-$(CONFIG_IMX7_PM_DOMAINS) += gpcv2.o
 obj-$(CONFIG_HAVE_IMX_MU) += mu/
 obj-$(CONFIG_ARCH_FSL_IMX8QM) += sc/
-obj-$(CONFIG_ARCH_FSL_IMX8QM) += pm-domains.o soc-imx8.o
+obj-$(CONFIG_ARCH_FSL_IMX8QM) += pm-domains.o
 obj-$(CONFIG_ARCH_FSL_IMX8MQ) += gpc-psci.o
+obj-$(CONFIG_HAVE_IMX_SOC) += soc-imx8.o
-- 
1.7.9.5

